bot/semester_switching.py

```diff
@@ -0,0 +1,325 @@
+            sorted_categories.append((category, index))
+            index += 1
+
+        # Now the voice category itself
+        sorted_categories.append((voice_category, index))
+        index += 1
+
+        # Now archived semester categories after voice category
+        for category in archived_categories:
+            sorted_categories.append((category, index))
+            index += 1
+
+        # And the rest of categories at the end
+        for category in second_half:
+            sorted_categories.append((category, index))
+            index += 1
+
+        # Final API position change of all categories
+        for category, index in sorted_categories:
+            await category.edit(position=index)
+
+    except discord.HTTPException as e:
+        raise Exception("Http error z discord API, error: " + str(e)) from e
+    except Exception as e:
+        raise Exception("Něco nastalo špatně během seřazování kategorii. " + str(e)) from None
+
+
+async def warn_user(ctx: discord.ApplicationContext) -> bool:
+    """This function will create a message with buttons to warn an administrator before using this command accidentally.
+    :param ctx: Slash command context
+    :return: True if command was accepted by administrator and False if declined
+    """
+    # Create an Event to signal when a button is pressed
+    event = asyncio.Event()
+
+    # default False if anything fails, we do not want to accept the command
+    accepted = False
+
+    # Button creation
+    accept_button = discord.ui.Button(style=discord.ButtonStyle.green, label="Přijmout")
+    decline_button = discord.ui.Button(style=discord.ButtonStyle.red, label="Odmítnout")
+
+    # Callbacks
+    async def accept_callback(interaction: discord.Interaction) -> None:  # pylint: disable=unused-argument
+        """
+        It is called when a user accepts the command via green button.
+        :param interaction: Interaction with button from user
+        :return: None
+        """
+        nonlocal accepted, message
+        accepted = True
+
+        event.set()
+
+    async def decline_callback(interaction: discord.Interaction) -> None:  # pylint: disable=unused-argument
+        """
+        It is called when a user declines the command via red button.
+        :param interaction: Interaction with button from user
+        :return: None
+        """
+        nonlocal accepted, message
+        accepted = False
+
+        event.set()
+
+    accept_button.callback = accept_callback
+    decline_button.callback = decline_callback
+
+    # View with buttons
+    view = discord.ui.View(timeout=30)  # 30 second time wait
+    view.add_item(accept_button)
+    view.add_item(decline_button)
+
+    # Final message
+    message: discord.Message | None = await ctx.followup.send(
+        "VAROVÁNÍ: Tento příkaz prohodí všechny semestry a přehází jejich role, "
+        "prosím, ujistěte se, že je nastaveno ID pro voice category a ID všech rolí studentů.",
```

Comment:
This message is implying that the roles of members, not the channels, will be switched. The message should be more clear about that.

It should also provide more specific technical requirements, which are written a few lines below in docstring, after all, it's not executed *that* often to make everyone bothered by its length. But in that case, it'd probably be important to add a little bit of markdown to the message to not drown the important parts.

Reply:
![image](https://github.com/user-attachments/assets/d9ec79cb-b77b-47a8-ba72-be70d29d0268)